src/options.ts

```diff
@@ -55,3 +60,32 @@ export const getOptions = (
   cache.set(cacheKey, compilerOptions);
   return compilerOptions;
 };
+
+export const getEnableExperimentalDecorators = (
+  ctx: TransformPluginContext, isTypeScript: boolean, cwd: string, key?: string | boolean
+) => {
+  if (!isTypeScript) return false;
+  const cacheKey = `${cwd}:${key ?? 'undefined'}`;
+  if (tsExperimentCache.has(cacheKey)) {
+    return tsExperimentCache.get(cacheKey) ?? false;
+  }
+  try {
+    // @ts-expect-error -- It's required to using 'import.mtea.url' but i don't want to change the tsconfig.
+    const tsPath = resolve('typescript/package.json', import.meta.url);
+    const { version } = JSON.parse(fs.readFileSync(fileURLToPath(tsPath), 'utf-8'));
```

Comment:
It seems that we are going to resolve `typescript/package.json` every time a file is transformed. The only thing that will change is the content of `tsconfig.json`, not the typescript version IMHO.

Reply:
I hoist it and remove useless code now. I also think that it is enough to check experimental decorators only once.